src/app/service/resource-type.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+import {Http} from '@angular/http';
+import 'rxjs/add/operator/toPromise';
+import {Constants} from '../constants';
+import {JsonApiDataStore} from 'jsonapi-datastore';
+import {ResourceType} from '../models/resource-type';
+
+@Injectable()
+export class ResourceTypeService {
+  private readonly resourceTypesUrl = Constants.BASE_URL + 'resource_types';
+
+  constructor(private http: Http) { }
+
+  private handleError(error: any): Promise<any> {
```

Comment:
Maybe you should make this a shared function sometime if this is how you want to handle all errors.